cmd/cloud-run/rotate-signify-certificate/rotate_signify_certificate.py

```diff
@@ -0,0 +1,279 @@
+"""Simple PubSub handler to rotate signify certificates"""
+
+import datetime
+import os
+import base64
+import json
+import sys
+import tempfile
+import traceback
+from typing import Any, Dict, List
+import requests
+from google.cloud import secretmanager
+from flask import Flask, Response, request, make_response
+from cryptography import x509
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.serialization import pkcs7, Encoding, PrivateFormat
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+app = Flask(__name__)
+project_id: str = os.getenv("PROJECT_ID", "")
+component_name: str = os.getenv("COMPONENT_NAME", "")
+application_name: str = os.getenv("APPLICATION_NAME", "")
+
+
+# TODO(kacpermalachowski): Move it to common package
+class LogEntry(dict):
+    """LogEntry simplifies logging by returning JSON string"""
+
+    def __str__(self):
+        return json.dumps(self)
+
+
+# pylint: disable-msg=too-many-locals
+@app.route("/", methods=["POST"])
+def rotate_signify_secret() -> Response:
```

Comment:
Break down rotate_signify_secret into smaller functions. For example:

parse_pubsub_message
get_old_certificate_and_key
generate_new_key_and_csr
fetch_new_certificate
update_secret

Reply:
It's divided into smaller functions